### ISSUE TITLE
Nothing can be taken for granted

### ISSUE THREAD

**Issue by nevrome** [poseidon-framework/poseidon-hs#8]
When working on the validate and the survey module I repeatedly stumbled upon the question which parts of a poseidon package are mandatory and which are optional. Of course this question is partly answered by the schema definition, but poet/trident will have to be able to perform its duties also on incomplete packages. 

Two main examples are "quick and dirty" modules for unpublished data without .janno + .bib files and the github repo check environment where the packages lack the genotype data.

I therefore suggest that `posPacGenotypeData` and `posPacJannoFile` in `PoseidonPackage` should be wrapped in a `Maybe`. All methods that work with packages should 1. check for missing input, including checks for file existence with `System.Directory.doesFileExist` and readability (except for Genotype data), 2. print useful data availability feedback for the user and 3. do what they're supposed to on the subset of data that fulfills all necessary prerequisites. 

Does this make sense?

**Comment by stschiff** [comment]
Hmm. Yes, I think this makes sense from an implementation point of view, so just within `poseidon-hs`. I mean... there's no harm if our implementation supports packages without Janno or Genotype files. I don't think, however, this should be officially declared somewhere. I think there's different levels of optionality. So, one suggestion might be that while we allow the code to parse packages without such files (printing out a warning perhaps), we would not officially declare these elements as "optional" in the package YAML file. What do you think?

**Comment by nevrome** [comment]
Ja - that was exactly what I had in mind. But making poet/trident very robust/versatile may work out better in the long term. In some ways it already is.

**Comment by nevrome** [comment]
I tried to implement my ideas for .janno and .bib files today and I'm fairly happy about the robustness I achieved. Interfaces like this

```
maybeLoadBibTeXFiles :: [PoseidonPackage] -> IO [Either PoseidonException (Either CiteprocException [Reference])]
maybeLoadJannoFiles :: [PoseidonPackage] -> IO [Either PoseidonException [Either PoseidonException PoseidonSample]]
```

are not pleasant on the eyes, but they do the trick.

I also did some experiments with the genotype data in that regard, but the arcane magic in `getJointGenotypeData` defeated me (for now). I have to leave this to you.
